.github/workflows/e2e-main.yml

```diff
@@ -156,3 +157,61 @@ jobs:
           ./cypress
           ./logs/test.log
         if-no-files-found: ignore
+
+  visual-regression-tests:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 45
+    needs: build
+    name: percy-screenshot-recording-${{ matrix.edition }}
+    env:
+      MB_EDITION: ${{ matrix.edition }}
+      ENTERPRISE_TOKEN: ${{ secrets.ENTERPRISE_TOKEN }}
+    strategy:
+      matrix:
+        java-version: [11]
+        edition: [ee]
+    services:
+      maildev:
+        image: maildev/maildev:1.1.0
+        ports:
+          - "80:80"
+          - "25:25"
+        credentials:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: Prepare front-end environment
+      uses: ./.github/actions/prepare-frontend
+    - name: Prepare JDK ${{ matrix.java-version }}
+      uses: actions/setup-java@v3
+      with:
+        java-version: ${{ matrix.java-version }}
+        distribution: 'temurin'
+    - name: Prepare Cypress environment
+      uses: ./.github/actions/prepare-cypress
+
+    - uses: actions/download-artifact@v2
+      name: Retrieve uberjar artifact for ${{ matrix.edition }}
+      with:
+        name: metabase-${{ matrix.edition }}-uberjar
+    - name: Get the version info
+      run: |
+        jar xf target/uberjar/metabase.jar version.properties
+        mv version.properties resources/
+    - name: Percy Test
+      run: |
+        CHANGED_SPECS=$(git diff --name-only origin/${{ github.base_ref }} origin/${{ github.head_ref }})
+        CYPRESS_SPECS=$(echo "$CHANGED_SPECS" | grep "metabase/scenarios")
+
+        if [ "$CHANGED_SPECS" == "" ]; then
+          echo "No files have been changed";
+        elif [ "$CHANGED_SPECS" == "$CYPRESS_SPECS" ]; then
+          echo "Not running Percy because the only changed files are Cypress E2E specs";
+        else
+          echo "Run Percy visual tests";
+          yarn run test-visual-run
+        fi
+      env:
+        PERCY_TOKEN: ${{ secrets.PERCY_TOKEN }}
+      if: env.PERCY_TOKEN != null && ${{ github.ref == 'refs/heads/master' }}
```

Comment:
@alxnddr can confirm or correct me if I'm wrong, but I understood that we should record only on `master`?

Reply:
> @alxnddr can confirm or correct me if I'm wrong, but I understood that we should record only on `master`?

Yep, you're right. Other jobs like `percy-visual-label` will record it on branches